packages/pages/src/common/src/function/internal/types.ts

```diff
@@ -50,19 +45,24 @@ export interface FunctionConfigInternal {
  */
 export const convertFunctionModuleToFunctionModuleInternal = (
   functionFilepath: path.ParsedPath,
-  functionModule: FunctionModule
+  functionModule: FunctionModule,
+  projectStructure: ProjectStructure
 ): FunctionModuleInternal => {
-  let functionInternal;
+  const serverlessSourcePath =
+    projectStructure.config.rootFolders.source +
+    "/" + // purposefully using string concatentation and not path.join
```

Comment:
can you explain why? It's because this is for importing modules right?

Reply:
What if the project structure has `\` instead? does that get properly handled? Maybe something on the project structure end to canonicalize things?

Reply:
This deals with the functions foldering to determine the function path/url. It's possible this is what broke Windows. If that's the case I'll fix in another PR. I didn't want to change the logic though of what was here before.

Why would the project structure have a `\`? You're saying if someone tried to do something custom? That seems weird unless they're doing something weird for Windows. I'm leaning towards not changing this as part of this change.

Reply:
> Why would the project structure have a \? You're saying if someone tried to do something custom?

We're treating the file structure as both the literal filepath and also as a valid npm import path. On windows, filepaths must use `\` but that'd be invalid for npm. I was just calling it out that if someone configured these paths on windows I think it'd break things.

Reply:
You're correct, but they should be setting them as npm paths, not local OS paths. We use path.resolve everywhere to handle fixing the slashes when necessary. I'll add validation for backslash in a future PR. I wonder if Vite validates this type of thing...nope. They just let it go through. So it's on the user to not do something silly.